### PR TITLE
Add feature queries support

### LINKED ISSUE #125: CSS Feature queries are mistakenly highlighted as syntax errors
For example:

bad-supports-error-highlighting-in-atom

The highlighting for a similar snippet tagged as a CSS file shows as expected:

screen shot 2016-05-17 at 10 33 57 am

Note that (nested) feature queries are <a href="https://github.com/sass/libsass/pull/626">fully supported</a> in SCSS as of a couple years ago. Would be nice to see this error go away. Thanks.

### PR DESCRIPTION
**Pull request by esdoppio**
closes #125

### CODE REVIEW

**Comment by esdoppio** [comment]
I've noticed there're many excessive escaping in `scss.cson`:

```cson
'logical_operators':
    'match': '\\b(\\=\\=|\\!\\=|\\<\\=|\\>\\=|\\<|\\>|not|or|and)\\b'
    'name': 'keyword.control.operator'
```

I mean, seriously dude. This is literally a backslash jungle there.

@50Wliu Can I open a PR to clean all those up? This kind of escaping is all over the file, it might cause conflicts to new PRs.

 How do u guys working on Atom communicate? Do you have something like an irc channel? I've got lots of questions to ask about developing for Atom.

(I'll delete this comment later if necessary.)

**Comment by 50Wliu** [comment]
> @50Wliu Can I open a PR to clean all those up? This kind of escaping is all over the file, it might cause conflicts to new PRs.

Yes please.

> How do u guys working on Atom communicate? I've got lots of questions to ask about developing for Atom.

https://discuss.atom.io/t/join-us-on-slack/16638?u=wliu

File: grammars/scss.cson [hunk 1]
```diff
@@ -542,6 +545,41 @@
         'include': '#selectors'
       }
     ]
+  'at_rule_supports':
+    'name': 'meta.at-rule.supports.scss'
+    'begin': '((@)supports\\b)'
```
Comment (50Wliu) [inline grammars/scss.cson]: \\b can go outside the capture group

File: grammars/scss.cson [hunk 2]
```diff
@@ -542,6 +545,41 @@
         'include': '#selectors'
       }
     ]
+  'at_rule_supports':
+    'name': 'meta.at-rule.supports.scss'
```
Comment (50Wliu) [inline grammars/scss.cson]: name before patterns

File: grammars/scss.cson [hunk 3]
```diff
@@ -542,6 +545,41 @@
...
+    'begin': '((@)supports\\b)'
+    'captures':
+      '1':
+        'name': 'keyword.control.at-rule.supports.scss'
+      '2':
+        'name': 'punctuation.definition.keyword.scss'
+    'end': '(?=\\{|$)'
```
Comment (50Wliu) [inline grammars/scss.cson]: { does not need to be escaped

**Comment by albell** [comment]
Hi, `@supports` isn't being highlighted as a syntax error anymore, but it still isn't highlighting as expected, or at least as I would expect. Property values inside the media feature should be recognized. And shouldn't `@supports` be purple in this theme, like `@media`?

feature-queries

Thanks.

**Comment by 50Wliu** [comment]
This is highlighting fine for me using the latest language-sass commit, 53966087daf0f0eaa276f615a922f02140079e14 (which happens to fix a bug with `@supports`).  That version should be available starting in Atom 1.11.0.

**Pull request merged**